app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/RequestEnableLocationServicesDialog.java

```diff
@@ -0,0 +1,43 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import org.kiwix.kiwixmobile.R;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+public class RequestEnableLocationServicesDialog extends DialogFragment {
+
+  public static final String TAG = "LocationDialog";
+
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+
+    AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+    builder.setMessage(R.string.request_enable_location)
```

Comment:
Is there a reason you are not using KiwixDialogShower instead of this DialogFragment?

Reply:
From what I understood, KiwixDialog.kt represents a regular dialog. Is it based on DialogFragment instead?
Both this and the WifiP2pServices... dialogs open another activity, so I wanted to avoid opening multiple such dialogs in case of multiple clicks

Reply:
DialogShower just shows an AlertDialog not in a fragment, I have found that DialogFragments are really mostly overblown and annoying, if I change my mind later on it will not be hard to convert KiwixDialogShower to use dialogFragments under the hood.
It should not be a concern that multiple clicks show multiple dialogs, once a click shows a dialog a dialog captures all screen input, a click outside the dialog will dismiss it and not propogate the event down to the underlying view hierarchy, a click on one of the buttons will automatically dismiss the dialog and perform the click action.

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/4b23a64f86e635e57d261965b385094308e5dd76#diff-e1d59ae6e15dcb502e125fb6e243d15dR487)